Sources/graphql-swift/codegen/jsonUtils.swift

```diff
@@ -0,0 +1,25 @@
+import Foundation
+
+public protocol InitializableFromJSON {
+    init(_ json: [String: Any])
```

Comment:
🎨   Idiomatic Swift would be to include the parameter label. I.e.

```suggestion
    init(json: [String: Any])
```

Reply:
Oh interesting. Thanks - this is exactly why I asked. I got the impression public labels were more for situations where there were overloads, with the labels being used to disambiguate. Is that not the case? Are named parameters preferred / standard in general?

Reply:
* `open` is for symbols that can be subclassed or overloaded that other modules should use.
* `public` is for read-only symbols that other modules should use.
* `internal` (default when nothing is specified) is basically `open` but only for the module itself.
* `fileprivate` is private to the file scope - so anything in that file can access it.
* `private` is what you think it is, private to that scope (file if at the top-level, type inside its scope)

And in general the less that’s `public` or `open` is generally considered good. For a binary (e.g. app or CLI tool) the only `public` symbols would be if the compiler forces you which usually only happens in certain specific extension+protocol conformance situations.

---

Named parameters are preferred and standard. There are specific cases where it’s preferred to omitting them but in general not. The official [API Design Guidelines](https://www.swift.org/documentation/api-design-guidelines/) document on Swift.org is well worth a read 👍